### PR TITLE
Remove RPH if Initialize returns error

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**
If RPH returns an error hresult, rather than failing Initialize altogether for CLRIE, this change will allow CLRIE to just remove the RPH instead.

### CODE REVIEW

File: src/InstrumentationEngine/ProfilerManager.cpp [hunk 1]
```diff
@@ -974,11 +974,16 @@ HRESULT CProfilerManager::Initialize(
...
-                IfFailRet(pCallback->Initialize((IUnknown*)(m_pWrappedProfilerInfo.p)));
+                IfFailLog(pCallback->Initialize((IUnknown*)(m_pWrappedProfilerInfo.p)));
             }
             else
             {
-                IfFailRet(pCallback->Initialize((IUnknown*)(m_pRealProfilerInfo.p)));
+                IfFailLog(pCallback->Initialize((IUnknown*)(m_pRealProfilerInfo.p)));
+            }
+
+            if (FAILED(hr))
```
Comment (wiktork) [inline src/InstrumentationEngine/ProfilerManager.cpp:984]: If RPH is turned on and fails, but no instrumentation methods are on, do we fail gracefully?

**Review by wiktork — APPROVED** [review]

File: src/InstrumentationEngine/ProfilerManager.cpp (hunk 1 above)
Comment (WilliamXieMSFT) [reply src/InstrumentationEngine/ProfilerManager.cpp:984]: I'll double check this, the desired behavior is that the process should have CLRIE running without any instrumentation methods.

Comment (WilliamXieMSFT) [reply src/InstrumentationEngine/ProfilerManager.cpp:984]: Confirmed that nothing happens in that case

**Pull request merged**